File: solidsim/pylib/pysolid.py
from SolidGeant4 import *

#----------------------------------------------------
# include exposed interfaces to the Python
# interpreter
#
# Example:
#class SolidGenericDet(G4VSensitiveDetector):
#    "Generic Solid SD"
#    def __init_(self, sdname):
#	G4VSensitiveDetector.__init__(self, sdname)
# 
#----------------------------------------------------

class SolidGenericDet(G4VSensitiveDetector, SolidData):
    "Generic Solid SD"
    def __init_(self, sdname):
	G4VSensitiveDetector.__init__(self, sdname)
	SolidData.__init__(self, sdname)

class SolidSimpleField(G4MagneticField, SolidSimpleField):
    "Generic Solid SD"
    def __init_(self, sdname, logvol):
	G4MagneticField.__init__(self)
	SolidFieldMap.__init__(self, sdname, logvol)

#class SolidData():
#    "Solid data container"
#    def __init_(self, sdname):

File: solidsim/pylib/pysolid.cc
#include <boost/python.hpp>

using namespace boost::python;

/*!
  Include exposed classes for the python interface

Example:
#include <G4VSensitiveDetector.hh>
#include <SolidGenericDet.hh>

BOOST_PYTHON_MODULE(pysolid) {
    class_<SolidGenericDet, bases<G4VSensitiveDetector> >("SolidGenericDet", init<const G4String&>() )
    ;
}

*/ 


#include <G4VSensitiveDetector.hh>
#include <G4LogicalVolume.hh>
#include <SolidGenericDet.hh>
#include <SolidData.hh>
#include <SolidSimpleField.hh>
#include <SolidFieldMap.hh>

BOOST_PYTHON_MODULE(pysolid) {
    class_<SolidGenericDet, bases<G4VSensitiveDetector> >("SolidGenericDet", init<const G4String&>() );
    class_<SolidSimpleField, bases<G4MagneticField> >("SolidSimpleField", init<G4String,G4LogicalVolume *>() );
//    class_<SolidData>("SolidGenericDet", init<const G4String&>() )
//    ;
}
